crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -224,6 +225,23 @@ impl Analyzer<'_, '_> {
                             .expand_intrinsic_types(actual_span, i)
                             .context("tried to expand intrinsic type as a part of normalization")?;
 
+                        let ctx = Ctx {
+                            in_actual_type: opts.in_type_or_type_param,
+                            ..self.ctx
+                        };
+
+                        if let Type::StringMapping(ty) = ty.normalize() {
```

Comment:
Can you move this logic to `expand_intrinsic_types`?

Reply:
Just did, took the liberty of adding `AssignOpts` to `expand_intrinsic_types` as a param so I can pass `opts.span`

Reply:
The latter, but I'm not 100% sure